File: tests/machine/x/typescript/order_by_map.ts
const data = [
  {a: 1, b: 2},
  {a: 1, b: 1},
  {a: 0, b: 5}
];
const sorted = data.slice().sort((a,b)=> {a: a.a, b: a.b} < {a: b.a, b: b.b} ? -1 : {a: a.a, b: a.b} > {a: b.a, b: b.b} ? 1 : 0);
console.log(sorted);

File: tests/machine/x/dart/order_by_map.dart
import 'dart:convert';

var data = [
  {'a': 1, 'b': 2},
  {'a': 1, 'b': 1},
  {'a': 0, 'b': 5},
];

var sorted = (() {
  var _q0 = <dynamic>[];
  for (var x in data) {
    _q0.add([
      {'a': x['a'], 'b': x['b']},
      x,
    ]);
  }
  _q0.sort(
    (a, b) => (jsonEncode(a[0]) as Comparable).compareTo(jsonEncode(b[0])),
  );
  _q0 = [for (var x in _q0) x[1]];
  return _q0;
})();

void main() {
  print(sorted);
}
